pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -508,3 +600,46 @@ func verifyNoConfigMapExists(c client.Client) bool {
 	// if we got here, it's good.  list was empty or everything passed
 	return true
 }
+
+// verifySecretExists verifies that the secret which referenced by the SyncSet exists in the test namespace
+func verifySecretExists(c client.Client, expected *SecretEntry) bool {
+	secret := corev1.Secret{}
+	err := c.Get(context.TODO(),
+		types.NamespacedName{Name: expected.name, Namespace: testNamespace},
+		&secret)
+
+	if err != nil {
+		return false
+	}
+
+	if expected.name != secret.Name {
+		return false
+	}
+
+	if expected.pagerdutyKey != string(secret.Data["PAGERDUTY_KEY"]) {
+		return false
+	}
+
+	return true
+}
+
+// verifyNoSecretExists verifies that the secret which referred by SyncSet does not exist
+func verifyNoSecretExists(c client.Client, expected *SecretEntry) bool {
+	secretList := &corev1.SecretList{}
+	opts := client.ListOptions{Namespace: testNamespace}
+	err := c.List(context.TODO(), &opts, secretList)
```

Comment:
Is there a reason we would do a List instead of a Get?

Reply:
Just followed the desigh for the syncset and the configmap which verify the non exists. It is better to keep them the same?